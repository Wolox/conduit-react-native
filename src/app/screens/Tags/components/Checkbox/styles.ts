import { StyleSheet } from 'react-native';
import { scale } from '@utils/scalingUtils';
import { green, lightGray } from '@constants/colors';
import { SPACING } from '@constants/dimentions';

const ICON_SIZE = scale(SPACING.BIG);
const CONTAINER_SIZE = scale(SPACING.MEDIUM);

export default StyleSheet.create({
  container: {
    width: '48%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: SPACING.XXSMALL,
    padding: SPACING.XSMALL,
    borderRadius: SPACING.XSMALL,
    borderColor: lightGray,
    borderWidth: 2
  },
  iconContainer: {
    width: CONTAINER_SIZE,
    height: CONTAINER_SIZE,
    alignItems: 'center',
    justifyContent: 'center'
  },
  icon: {
    width: ICON_SIZE,
    height: ICON_SIZE,
    tintColor: green
  },
  selected: {
    backgroundColor: lightGray
  }
});
