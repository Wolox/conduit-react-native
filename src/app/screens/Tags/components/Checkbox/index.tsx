import React from 'react';
import { TouchableOpacity, Image, View } from 'react-native';
import CustomText from '@components/CustomText';
import icCheck from '@assets/icons/icCheck.png';

import styles from './styles';

interface Props {
  option: string;
  selected: boolean;
  onPress: (option: string) => void;
}

function Checkbox({ selected = false, option, onPress }: Props) {
  const handlePress = () => onPress(option);
  const isSelected = selected;
  return (
    <TouchableOpacity style={[styles.container, isSelected && styles.selected]} onPress={handlePress}>
      <CustomText>{option}</CustomText>
      <View style={[styles.iconContainer]}>
        {isSelected && <Image source={icCheck} style={styles.icon} />}
      </View>
    </TouchableOpacity>
  );
}

export default Checkbox;
