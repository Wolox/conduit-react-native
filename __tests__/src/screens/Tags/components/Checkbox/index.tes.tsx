import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '@screens/Tags/components/Checkbox';

describe('Checkbox', () => {
  test('Checkbox Snapshot', () => {
    const onPress = (_option: string) => null;
    const item = { selected: false, option: '', onPress };
    const tree = renderer.create(<Checkbox {...item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
