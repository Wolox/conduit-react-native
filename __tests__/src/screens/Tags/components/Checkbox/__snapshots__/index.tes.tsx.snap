// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checkbox Snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#E5E5E5",
      "borderRadius": 6,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 4,
      "opacity": 1,
      "padding": 6,
      "width": "48%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#000",
          "fontFamily": "Nunito",
          "fontSize": 25.142857142857142,
          "fontStyle": "normal",
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Nunito",
            "fontSize": 25.142857142857142,
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ],
        undefined,
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 34.285714285714285,
          "justifyContent": "center",
          "width": 34.285714285714285,
        },
      ]
    }
  />
</View>
`;
